docs/en/operations/settings/settings-formats.md

```diff
@@ -1951,10 +1951,34 @@ The maximum allowed size for String in RowBinary format. It prevents allocating
 
 Default value: `1GiB`.
 
+### output_format_binary_encode_types_in_binary_format {#output_format_binary_encode_types_in_binary_format}
+
+Write data types in [binary format](../../sql-reference/data-types/data-types-binary-encoding.md) instead of type names in RowBinaryWithNamesAndTypes output format.
+
+Disabled by default.
+
+### input_format_binary_decode_types_in_binary_format {#input_format_binary_decode_types_in_binary_format}
```

Comment:
I wonder if the new settings are only useful for testing but not for productive use outside the Dynamic datatype (if so, we should document this)? As a user, how would I decide?

(It would be ideal if the database could determine the format of the types, binary vs. string, automatically and do the right thing without needing a setting, but that is probably infeasible.)

Reply:
We don't want to use binary encoding in server-server or native client-server communications, because it's not that useful. These settings are needed only for other clients in other languages that can use this encoding in the future. 

The main usage of this encoding is still new JSON data type, but we decided to add these settings for possible future usage.

I will add binary output format of Dynamic type to Dynamic documentation.